src/coredump/coredump.c

```diff
@@ -1023,84 +1047,88 @@ static int send_iovec(const struct iovec_wrapper *iovw, int input_fd) {
+                                       argc, _META_ARGV_MAX);
 
-        r = get_process_exe(pid, &context[CONTEXT_EXE]);
-        if (r < 0)
-                log_warning_errno(r, "Failed to get EXE, ignoring: %m");
+        for (i = 0; i < _META_ARGV_MAX; i++) {
 
-        if (cg_pid_get_unit(pid, &context[CONTEXT_UNIT]) >= 0) {
-                if (!is_journald_crash((const char**) context)) {
-                        /* OK, now we know it's not the journal, hence we can make use of it now. */
-                        log_set_target(LOG_TARGET_JOURNAL_OR_KMSG);
-                        log_open();
-                }
+                t = argv[i];
 
-                /* If this is PID 1 disable coredump collection, we'll unlikely be able to process it later on. */
-                if (is_pid1_crash((const char**) context)) {
-                        log_notice("Due to PID 1 having crashed coredump collection will now be turned off.");
-                        disable_coredumps();
+                switch (i) {
+                case META_ARGV_TIMESTAMP:
+                        /* The journal fields contain the timestamp padded with six
+                         * zeroes, so that the kernel-supplied 1s granularity timestamps
+                         * becomes 1µs granularity, i.e. the granularity systemd usually
+                         * operates in. */
+                        t = free_timestamp = strjoin(argv[i], "000000");
+                        if (!t)
+                                return log_oom();
+                        break;
+                case META_ARGV_SIGNAL:
+                        /* For signal, record its pretty name too */
+                        if (safe_atoi(argv[i], &signo) >= 0 && SIGNAL_VALID(signo))
+                                iovw_put_string_field(iovw, "COREDUMP_SIGNAL_NAME=SIG",
+                                                      signal_to_string(signo));
+                        break;
+                default:
+                        break;
                 }
 
-                iovw_put_string_field(iovw, "COREDUMP_UNIT=", context[CONTEXT_UNIT]);
+                r = iovw_put_string_field(iovw, meta_field_names[i], t);
+                if (r < 0)
+                        return r;
         }
 
-        if (cg_pid_get_user_unit(pid, &t) >= 0)
-                iovw_put_string_field_free(iovw, "COREDUMP_USER_UNIT=", t);
-
-        /* The next few are mandatory */
-        r = iovw_put_string_field(iovw, "COREDUMP_PID=", context[CONTEXT_PID]);
-        if (r < 0)
-                return r;
+        /* Cache some of the process metadata we collected so far and that we'll need to
+         * access soon */
+        return save_context(context, iovw);
+}
 
-        r = iovw_put_string_field(iovw, "COREDUMP_UID=", context[CONTEXT_UID]);
-        if (r < 0)
-                return r;
+static int gather_pid_metadata_from_runtime(struct iovec_wrapper *iovw, Context *context) {
```

Comment:
as above: i think "from_proc" woud be more descriptive than "from_runtime", and explains precisely what this is

Reply:
see my previous reply, maybe simply drop "from_runtime" and name it "gather_pid_metadata()" ?

Reply:
sounds ok to me too.

Reply:
changed.